src/app/pokemons/[name]/__test__/page.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { render, screen } from '@testing-library/react';
+import Page from '../page';
+import { ReduxProvider } from '@/redux/provider';
+import { PokemonDetails } from '@/mocks/mock-data/PokemonDetails';
+
+describe('Pokemon Detail Page', () => {
+  const props = {
+    params: {
+      name: 'bulbasaur',
+    },
+  };
+
+  beforeEach(() => {
+    global.fetch = jest.fn().mockImplementationOnce(() =>
+      Promise.resolve({
+        json: () => Promise.resolve(PokemonDetails),
+      })
+    );
+  });
+
+  it('should display pokemon name', async () => {
+    const Result = await Page(props);
+    const children = Result;
+    render(<ReduxProvider children={children}></ReduxProvider>);
```

Comment:
Create a custom function that accepts a component and renders it with the reduxProvider. This way you can override the implementation of the render from RTL and use it directly thereby making the code more readable,